pyfar/classes/transmission_matrix.py

```diff
@@ -0,0 +1,796 @@
+        # Admittance form for Zl = inf
+        idx_inf, __, __ = self._check_for_inf(Zl)
+        nominator.freq[idx_inf] = (self.A + self.B / Zl).freq[idx_inf]
+        denominator.freq[idx_inf] = (self.C + self.D / Zl).freq[idx_inf]
+
+        # Avoid cases where denominator is zero, examples
+        # Zl = inf & C = 0; Zl = 0 & D = 0
+        denominator.freq[denominator.freq == 0] = np.finfo(float).eps
+        return nominator / denominator
+
+    def output_impedance(self, Zl: complex | FrequencyData) -> FrequencyData:
+        """Calculates the output impedance given the load impedance Zl at the
+        input.
+
+        Two-port representation::
+
+                    o---xxxxxxxxx---o
+                    |   x       x
+                    Zl  x       x      <--Zout
+                    |   x       x
+                    o---xxxxxxxxx---o
+
+        See Equation (2-6) in Reference [1]_:
+        :math:`Z_\\mathrm{out} = \\frac{DZ_L + B}{CZ_L + A}`
+
+        For a code example, see :py:func:`~input_impedance` and exchange
+        respective method call with `output_impedance`.
+
+        Parameters
+        ----------
+        Zl : scalar | FrequencyData
+            The load impedance data as scalar or FrequencyData. In latter case,
+            the shape must match the entries of the T-matrix, i.e.
+            shape(tmat.A.freq) == shape(Zl.freq), or must be broadcastable.
+
+        Returns
+        -------
+        Zout : FrequencyData
+            A FrequencyData object with the resulting output impedance. The
+            cshape is identical to the entries of the T-matrix, i.e.
+            tmat.A.cshape == Zout.cshape.
+
+        """
+        nominator = (self.D * Zl + self.B)
+        denominator = (self.C * Zl + self.A)
+
+        # Admittance form for Zl = inf
+        idx_inf, __, __ = self._check_for_inf(Zl)
+        nominator.freq[idx_inf] = (self.D + self.B / Zl).freq[idx_inf]
+        denominator.freq[idx_inf] = (self.C + self.A / Zl).freq[idx_inf]
+
+        # Avoid cases where denominator is zero, examples
+        # Zl = 0 & A = 0; Zl = inf & C = 0
+        denominator.freq[denominator.freq == 0] = np.finfo(float).eps
+        return nominator / denominator
+
+    def transfer_function(self, quantity_indices,
+                          Zl: complex | FrequencyData) -> FrequencyData:
+        """Returns the transfer function (output/input) for specified
+        quantities and a given load impedance.
+
+        The transfer function (TF) is the relation between an output and input
+        quantity of modelled two-port and depends on the load impedance at the
+        output. Since there are two quantities at input and output
+        respectively, four transfer functions exist in total. The first usually
+        refers to the "voltage-like" quantity (:math:`Q_1`) whereas the second
+        refers to the "current-like" quantity (:math:`Q_2`).
+
+        The TFs can be derived from Equation (2-1) in Reference [1]_:
```

Comment:
I don't think it is necessary to introduce the abbreviation "TF" for this docstring. Could please replace it with transfer function? :)